horace_core/spaghetti_plot.m

```diff
@@ -109,426 +113,486 @@
+        wdisp(i) = cut(sqw_in, proj, u1, u2, u3, ebin); % Keep pixels for 1D cuts
         u1v = u1(1):u1(2):u1(3);
-        for j=1:numel(u1v)-1
-            if wdisp(i).data.pix.num_pixels > 0
-                varargout{2}{i}(j) = d1d(cut(wdisp(i),u1v(j:j+1),[]));
-            else
-                varargout{2}{i}(j) = d1d();
+
+        if wdisp(i).data.pix.num_pixels > 0
+            for j=1:numel(u1v)-1
+                varargout{2}{i}(j) = cut(wdisp(i), u1v(j:j+1), []);
+            end
+
+            if ~opt.withpix
+                varargout{2}{i} = cellfun(@d1d, varargout{2}{i});
             end
+
+        else
+            varargout{2}{i} = repmat(d1d(), 1, numel(u1v)-1);
         end
+
         wdisp(i) = d2d(wdisp(i));
     else
-        wdisp(i) = cut_sqw(sqw,proj,u1,u2,u3,ebin,'-nopix');
+
+        wdisp(i) = cut(sqw_in, proj, u1, u2, u3, ebin, '-nopix');
     end
+
     if present.labels
-        titlestr = sprintf('Segment from "%s" (%f %f %f) to "%s" (%f %f %f)',...
-            opt.labels{i},rlp(i,:),opt.labels{i+1},rlp(i+1,:));
+        titlestr = sprintf('Segment from "%s" (%f %f %f) to "%s" (%f %f %f)', ...
+                           opt.labels{i}, rlp(i, :), opt.labels{i+1}, rlp(i+1, :));
     else
-        titlestr = sprintf('Segment from (%f %f %f) to (%f %f %f)',rlp(i,:),rlp(i+1,:));
+        titlestr = sprintf('Segment from (%f %f %f) to (%f %f %f)', ...
+                           rlp(i, :), rlp(i+1, :));
     end
-    if i>1
+
+    if i > 1
         wdisp(i).title = titlestr;
     else
-        binstr = sprintf(['q bin size approximately %f ',char(197),'^{-1}'],opt.qbin);
+        binstr = sprintf(['q bin size approximately %f ', char(197), '^{-1}'], opt.qbin);
         if numel(opt.qwidth)==1
-            wdisp(i).title=sprintf(['integrated over %f ',char(197),...
-                '^{-1} in perpendicular q-directions\n%s\n%s'],opt.qwidth(1),binstr,titlestr);
+            wdisp(i).title=sprintf(['integrated over %f ', char(197), ...
+                '^{-1} in perpendicular q-directions\n%s\n%s'], opt.qwidth(1), binstr, titlestr);
         else
-            wdisp(i).title=sprintf(['integrated over %f ',char(197),...
-                '^{-1} in qv and %f ',char(197),'^{-1} in qw\n%s\n%s'],opt.qwidth,binstr,titlestr);
+            wdisp(i).title=sprintf(['integrated over %f ', char(197), ...
+                '^{-1} in qv and %f ', char(197), '^{-1} in qw\n%s\n%s'], opt.qwidth, binstr, titlestr);
         end
     end
+
+
 end
 
-%-------------------------
-if nargout>0
+if nargout > 0
     varargout{1}=wdisp;
 end
 
-
 % Plot dispersion
 %----------------
 if nargout<1 || (~present.noplot)
     plot_dispersion(wdisp,opt)
 end
 
+end
 
 %========================================================================================================
 function plot_dispersion(wdisp_in,opt)
 % Plots the dispersion in the structure wdisp
-persistent old_matlab;
-if isempty(old_matlab)
-    old_matlab = verLessThan('Matlab','8.6');
```

Comment:
What's the oldest version of Matlab we now support?

Reply:
2018b